notify/opsgenie/opsgenie.go

```diff
@@ -189,14 +192,19 @@ func (n *Notifier) createRequest(ctx context.Context, as ...*types.Alert) (*http
 	return req.WithContext(ctx), true, nil
 }
 
-func (n *Notifier) retry(statusCode int) (bool, error) {
-	// https://docs.opsgenie.com/docs/response#section-response-codes
-	// Response codes 429 (rate limiting) and 5xx are potentially recoverable
-	if statusCode/100 == 5 || statusCode == 429 {
-		return true, fmt.Errorf("unexpected status code %v", statusCode)
-	} else if statusCode/100 != 2 {
-		return false, fmt.Errorf("unexpected status code %v", statusCode)
+func (n *Notifier) retry(statusCode int, body io.Reader) (bool, error) {
+	if statusCode/100 == 2 {
+		return false, nil
 	}
 
-	return false, nil
+	err := errors.Errorf("unexpected status code %v", statusCode)
```

Comment:
I think we put too many things into this function to just name it `retry` (: 

Can we have `isRetriable` with what we have previously and then have this:
```
err := errors.Errorf("unexpected status code %v", statusCode)
if body != nil {
		if bs, errRead := ioutil.ReadAll(body); errRead == nil {
			err = errors.Errorf("%s: %s", err, string(bs))
		}
	}
```
in `Notify` if the `isRetriable(..)` is false?

Reply:
Hmm, it wouldn't work as-is because we can have errors that shouldn't be retried and for which logging the response body is still useful IMO.

FWIW the current form is consistent with what we have in other places (eg [Slack](https://github.com/prometheus/alertmanager/blob/4706dc7e6aebb486aeeac8c385905acc8c32fcd6/notify/slack/slack.go#L183-L198)). If anything we should factor out what is common across notifiers.